### PR TITLE
Rename the config package to gitconfig

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph I finally found some time to work undistracted on Git Town. I had to physically sit on a plane without wifi. I am sending a number of PRs with structural improvements. 

This PR renames the `github.com/Originate/git-town/lib/config` package to `.../gitconfig`. This has several reasons:
- `config` is a too generic name. I want to use it for configuration variables and structs later
- this package really only is concerned with reading and writing configuration out of Git metadata
- in the future we might allow providing additional configuration information via the command line

### CODE REVIEW

**Comment by charlierudolph** [comment]
I don't like `gitconfig` since we have a `git` folder as well. I'd rather we just move the config file into the git folder.

**Comment by kevgo** [comment]
Hmm, I don't understand your comment. Do you want to merge the `git` and `gitconfig` package into one called `git`? The former has mostly methods that I could see in the latter, like
- HasRemote
- GetMainBranch
- EnsureIsFeatureBranch
- GetChildBranches
etc.

Maybe we start by moving these methods into the `git` package?

**Comment by charlierudolph** [comment]
Yes, I am saying merge the gitconfig and git pacakages.

**Comment by kevgo** [comment]
All comments addressed

File: lib/git/gitconfig.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package config
+package git
```
Comment (charlierudolph) [inline lib/git/gitconfig.go]: Can this file be renamed to config.go?

**Pull request merged**